users/postgres/users.go

```diff
@@ -66,18 +67,36 @@ func (ur userRepository) RetrieveByID(_ context.Context, email string) (users.Us
 type dbUser struct {
 	Email    string `json:"email"`
 	Password string `json:"password"`
+	Metadata string `json:"metadata",omitempty`
 }
 
 func toDBUser(u users.User) dbUser {
+	m := "{}"
+	if len(u.Metadata) > 0 {
+		data, err := json.Marshal(u.Metadata)
+		if err != nil {
+			return dbUser{}
```

Comment:
Wee do not log anything and we do not return error to the caller? I suppose we expect logging middleware to log, but it must have some indication that error happened.